### ISSUE TITLE
Arbitrary type converter

### ISSUE THREAD

**Issue by Maratori** [cemdervis/SharpConfig#46]
It would be great to add possibility to convert from/to string an arbitrary object type.
As an example it could be done by changing FallbackStringConverter as following.

```c#
internal sealed class FallbackStringConverter : ITypeStringConverter
{
    public string ConvertToString(object value)
    {
        try {
            return TypeDescriptor.GetConverter(value).ConvertTo(value, typeof(string)) as string;
        }
        catch
        {
            //
        }
        return value.ToString();
    }

    public object ConvertFromString(string value, Type hint)
    {
        try
        {
            return TypeDescriptor.GetConverter(hint).ConvertFrom(value);
        }
        catch
        {
            //
        }
        throw new NotImplementedException();
    }

    public Type ConvertibleType
    {
        get { return null; }
    }
}
```

Also method CreateObjectFromString should be changed in following part not to throw an exception.

```c#
private static object CreateObjectFromString(string value, Type dstType)
{
    ................
    var converter = Configuration.FindTypeStringConverter(dstType);
    if (converter == Configuration.FallbackConverter)
    {
        // The fallback converter is not able to create arbitrary objects
        // from strings, so this means that there is no type converter
        // registered for dstType.
        throw SettingValueCastException.CreateBecauseConverterMissing(value, dstType);
    }
    .......................
}
```

**Comment by cemdervis** [comment]
Hey there, sorry for the late reply.
Do you have specific use cases for which TypeDescriptor would be necessary?
I'd have to test how well TypeDescriptor integrates into SharpConfig.

**Comment by Maratori** [comment]
For example, I'd like to store size of some controls in WPF application. It has a type `System.Windows.GridLength`. And I don't want to create custom converter where it isn't needed.

**Comment by cemdervis** [comment]
Alright, seems like a good addition. I will implement it soon, thanks.

**Comment by cemdervis** [comment]
Fixed with commit 0b3fb2a. Thanks!
